src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -533,6 +533,15 @@ private function getArgumentsAsPhp(\DOMElement $node, $name, $file, $lowercase =
                         throw new InvalidArgumentException(sprintf('Tag "<%s>" with type="service_locator" only accepts maps of type="service" references in "%s".', $name, $file));
                     }
                     break;
+                case 'tagged_locator':
+                    if (!$arg->getAttribute('tag') && !$arg->getAttribute('index-by')) {
```

Comment:
not sure about the "index-by" check, we don't have it for iterators and I don't think it is legit

Reply:
As we are not just injecting tagged service like `!tagged` do if we only specify the tag, I think here the `!tagged_locator` really needs at minima the index-by attribute to do his "special" behavior. 

Otherwise, we would use the `!service_locator`

Reply:
OK, then it's a `||`